code/game/atoms_movable.dm

```diff
@@ -375,18 +375,29 @@
 	loc = new_loc
 	Moved(old_loc)
 
-////////////////////////////////////////
-// Here's where we rewrite how byond handles movement except slightly different
-// To be removed on step_ conversion
-// All this work to prevent a second bump
-/atom/movable/Move(atom/newloc, direction, glide_size_override = 0)
-	. = FALSE
-	if(!newloc || newloc == loc)
-		return
+/atom/movable/Move(atom/newloc, direction, glide_size_override)
+	var/atom/movable/pullee = pulling
+	if(!moving_from_pull)
+		check_pulling()
+	if(!loc || !newloc || loc == newloc)
+		return FALSE
 
 	if(!direction)
 		direction = get_dir(src, newloc)
 
+	var/can_pass_diagonally = FALSE
+	if ((direction & (direction - 1))) //Check if the first part of the diagonal move is possible
+		if(direction & NORTH)
+			can_pass_diagonally = get_step(loc, NORTH)?.Enter(src) ? NORTH : 0
```

Comment:
Use NONE instead of 0 here. Also I'm pretty sure you can do this more efficiently by storing which two orthogonal directions the movement direction is composed of and then checking the north/south one then the east/west one in this manner. Should be profiled though